rpmlb/builder/base.py

```diff
@@ -94,27 +95,40 @@ def build_with_retrying(self, package_dict, **kwargs):
     def build(self, package_dict, **kwargs):
         raise NotImplementedError('Implement this method.')
 
-    def edit_spec_file(self, spec_file):
-        spec_file_origin = '{0}.orig'.format(spec_file)
-        os.rename(spec_file, spec_file_origin)
-        fh_r = None
-        fh_w = None
-        try:
-            fh_r = open(spec_file_origin, 'r')
-            fh_w = open(spec_file, 'w')
-            fh_w.write('# Edited by rpmlb\n')
-            yield(fh_r, fh_w)
-        finally:
-            if fh_w:
-                fh_w.close()
-            if fh_r:
-                fh_r.close()
+    @staticmethod
+    @contextmanager
+    def edit_spec_file(target: Path):
+        """Safely edit a SPEC file in-place.
+
+        The target is backed up as '{target}.orig' if needed.
+
+        Keyword arguments:
+            target: The modified SPEC file path.
+
+        Returns:
+            Context manager providing open handles
+            for input and output file.
+        """
+
+        # Ensure path type
+        target = Path(target) if not isinstance(target, Path) else target
+
+        # Back up the original
+        source = target.with_suffix('.spec.orig')
+        if not source.exists():
+            target.rename(source)
+
+        # Provide the handles
+        with source.open(mode='r') as src, target.open(mode='w') as tgt:
```

Comment:
After talking with mhroncok on IRC, let me comment. Other things look good.

## 1.
Variable `source` variable is an object of `Path` class. Variable `src` variable is an object of `File` class(?) And `src` is shorten word of `source`.
Variable `target` is an object of `Path` class. Variable `tgt` is an object of `File` class? And `tgt` is shorten word of `target`.

Using same concept (meaning)'s word for different class is usual in Python world?
It is not usual for me in terms of naming consistency. (1 kind of naming for 1 concept).

`target` as `Path` object, `target_file` as `File(?)` object looks better isn't it?

Below sample using `p` as `Path`, `f` as `File(?)` as reference.
https://docs.python.org/3/library/pathlib.html#pathlib.Path.open

## 2.
About using shorten word, which case is popular for Python world?

a. Use not shorten word as variable as much as possible such as `target`. Use shorten word for known word as exception `tgt`.
b. Use shorten word as variable usually.  

My understanding is for example,
Perl (Modern), Ruby, Java, C#, JavaScript, PHP are case a.
C, Perl (Traditional) are case b.